backend.go

```diff
@@ -101,14 +108,47 @@ func Backend() *kubeAuthBackend {
 			},
 			pathsRole(b),
 		),
+		InitializeFunc: b.initialize,
 	}
 
+	// Set default HTTP client
+	b.httpClient = cleanhttp.DefaultPooledClient()
+
 	// Set the review factory to default to calling into the kubernetes API.
 	b.reviewFactory = tokenReviewAPIFactory
 
 	return b
 }
 
+// initialize is used to handle the state of config values just after a plugin has been mounted
+func (b *kubeAuthBackend) initialize(ctx context.Context, req *logical.InitializationRequest) error {
+	// Try to load the config on init
+	config, err := b.loadConfig(ctx, req.Storage)
+	if err != nil {
+		return err
+	}
+	if config == nil {
```

Comment:
Updated in https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/142/commits/c1230008498ef59342ab8495ffe33176de931ace. This allows other functions to return an error if the config is nil and this will allow us in our initialize fucntion to check if we have a nil config value and just continue as normal.